### ISSUE TITLE
Salt overlapping bond

### ISSUE THREAD

**Issue by baoilleach** [cdk/depict#26]
Salt placement is usually great but I came across the following where the Br- overlaps a bond.
```
C=C(C[N+](C)(C)C)OC(=O).[Br-]
```
![png](https://user-images.githubusercontent.com/45337/96236220-a7843280-0f93-11eb-9902-1bf1fdf5a9df.png)

**Comment by johnmay** [comment]
Thanks very hard to think of a simple fix but will have a think. Basically the way it works is adds in a "longer" bond that isn't shown.

**Comment by johnmay** [comment]
Oh actually perhaps for nonterminal X>3 I should just put it near the closest terminal bond.

**Comment by egonw** [comment]
OT: btw, is the OHC abbreviation common?

**Comment by johnmay** [comment]
Yes it's reverse CHO... Things do get confusing though - RDKit recently added abbreviations and had some confusion examples where they used half forward, half backwards "CH3(O)NH-" (i think) - this makes it look like the O is connected to the first Carbon. We would make that "H3C(O)HN-" which matches ChemDraw and is the most logical way to do it - otherwise it gets very messy. For RDKit's way it would be valid if you wrote "CH3NH(O)-" but then it gets hard to join on to the Nitrogen since it's no longer at the end of the formula  CH3NHO- would make it look like it joins on the O. 

Here is a screenshot from ChemDraw - I drew the top and rotated it 180.

![image](https://user-images.githubusercontent.com/983232/96385820-41173400-118e-11eb-8a0c-939968acfa92.png)

**Comment by johnmay** [comment]
Reopen - don't think it's fixed yet

**Comment by johnmay** [comment]
I linked the wrong issue :-)
